apps/src/lab2/views/components/Instructions/ResourcePanel/useValidationTour.tsx

```diff
@@ -0,0 +1,270 @@
+import {Steps} from 'intro.js-react';
+import React, {useMemo, useState, useEffect} from 'react';
+
+import {Tabs} from '@cdo/apps/lab2/types';
+import {ValidationSettings} from '@cdo/apps/lab2/views/components/Instructions/InstructionsV2';
+import {
+  PYTHONLAB_VALIDATION_TOUR_SEEN,
+  PYTHONLAB_RESOURCE_PANEL_ONBOARDING_TOUR_SEEN,
+} from '@cdo/apps/pythonlab/constants';
+import {commonI18n} from '@cdo/apps/types/locale';
+import {tryGetLocalStorage, trySetLocalStorage} from '@cdo/apps/utils';
+
+import {VALIDATION_TOUR_STEPS} from './validationTourHelpers';
+
+interface UseValidationTourProps {
+  appName: string;
+  hasValidationConditions: boolean;
+  validationSettings: ValidationSettings | undefined;
+  setCurrentTab: (tab: Tabs) => void;
+  onValidate: (() => void) | undefined;
+}
+
+// Currently this hook is only used for Python Lab.
+// If other labs would like to opt in to use this hook, we can update the hook to work for their labs.
+export const useValidationTour = ({
+  appName,
+  hasValidationConditions,
+  validationSettings,
+  setCurrentTab,
+  onValidate,
+}: UseValidationTourProps) => {
+  const isPythonLab = appName === 'pythonlab';
+  const [validationTourEnabled, setValidationTourEnabled] = useState(false);
+  const [validationTourStep, setValidationTourStep] = useState(0);
+  const validationTabEnum = Tabs.Validation;
+  const pythonlabValidationTourSeen = tryGetLocalStorage(
+    PYTHONLAB_VALIDATION_TOUR_SEEN,
+    'no'
+  );
+  const pythonlabOnboardingTourSeen = tryGetLocalStorage(
+    PYTHONLAB_RESOURCE_PANEL_ONBOARDING_TOUR_SEEN,
+    'no'
+  );
+
+  // The Done button on the third step (index 2) is always enabled.
+  // The Next button on the first two steps (indexes 0 and 1) is disabled until the user completes an action.
+  const [validationTourStepsEnabled, setValidationTourStepsEnabled] = useState([
+    false,
+    false,
+    true,
+  ]);
+
+  // Enable validation tour if conditions are met.
+  useEffect(() => {
+    if (!isPythonLab) return;
+
+    const shouldShowValidationTour =
+      validationSettings &&
+      hasValidationConditions &&
+      pythonlabValidationTourSeen !== 'yes' &&
+      pythonlabOnboardingTourSeen === 'yes'; // If user hasn't seen both tours, show onboarding tour first.
```

Comment:
if the user has not yet seen the onboarding tour, then the onboarding tour will first be shown. Once it is seen, then the validation tour will be shown.

Reply:
Will it be shown immediately or on next load?

Reply:
Immediately, not on next load:

https://github.com/user-attachments/assets/73c6e363-28b9-41b3-bc9f-cc239b7590c9